lms/services/canvas_api/client.py

```diff
@@ -255,6 +255,24 @@ class _PublicURLSchema(RequestsResponseSchema):
 
         public_url = fields.Str(required=True)
 
+    def course_group_categories(self, course_id):
+        return self._client.send(
+            "GET",
+            f"courses/{course_id}/group_categories",
+            schema=self._ListGroupCategories,
+        )
+
+    class _ListGroupCategories(RequestsResponseSchema):
+        many = True
+
+        id = fields.Integer(required=True)
+
+        name = fields.Str(required=True)
+        description = fields.String(default=None, allow_none=True)
+
+        role = fields.Str(required=False, allow_none=True)
+        content_type = fields.Str(required=False, allow_none=True)
```

Comment:
Do we need `description`, `role` and `content_type` here? I don't think we're using them so I think we can just omit them from the schema and they'll be ignored. This works because we configure Marshmallow to ignore unknown fields: https://github.com/hypothesis/lms/blob/d220299bd95b8f4d0262a45c169e35ab97948a1e/lms/validation/_base.py#L45-L46

Reply:
Yup, those were useless at the moment. Removed.